cmake/icudata.cmake

```diff
@@ -98,17 +98,38 @@ if (ICU_PKGDATA_EXECUTABLE)
     generate_icudata_resource_bundle(curr ${_ICUDATA_BINARY_DIR}/curr)
     generate_icudata_resource_bundle(unit ${_ICUDATA_BINARY_DIR}/unit)
 
-    if (APPLE)
+    if (CMAKE_CROSSCOMPILING AND CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
```

Comment:
if ((APPLE) OR (CMAKE_CROSSCOMPILING AND CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")) ??